### PR TITLE
DL-10873 - improved logging

### PR DESCRIPTION
**Pull request by derickdamoah**

### CODE REVIEW

File: app/connectors/httpParsers/CustomerDetailsHttpParser.scala [hunk 1]
```diff
@@ -21,35 +21,36 @@ import models.CustomerDetails
...
     override def read(method: String, url: String, response: HttpResponse): HttpResult[CustomerDetails] = {
+      implicit val res: HttpResponse = response
 
       response.status match {
         case OK =>
-          logger.debug("[CustomerCircumstancesHttpParser][read]: Status OK")
+          debug("[CustomerCircumstancesHttpParser][read]: Status OK")
           response.json.validate[CustomerDetails].fold(
             invalid => {
-              logger.debug(s"[CustomerCircumstancesHttpParser][read]: Invalid Json - $invalid")
```
Comment (scott-goodwin) [inline app/connectors/httpParsers/CustomerDetailsHttpParser.scala:37]: just check invalid isnt returning any PII please

File: app/connectors/httpParsers/CustomerDetailsHttpParser.scala [hunk 2]
```diff
@@ -21,35 +21,36 @@ import models.CustomerDetails
...
             valid => Right(valid)
           )
 
         case PRECONDITION_FAILED =>
-          logger.debug(s"[CustomerCircumstancesHttpParser][read]: Status $PRECONDITION_FAILED")
+          errorLogRes(s"[CustomerCircumstancesHttpParser][read]: Status $PRECONDITION_FAILED")
           Left(Migration)
 
         case NOT_FOUND =>
-          logger.debug(s"[CustomerCircumstancesHttpParser][read]: Status $NOT_FOUND")
```
Comment (scott-goodwin) [inline app/connectors/httpParsers/CustomerDetailsHttpParser.scala:48]: same here should be a warn

File: app/connectors/httpParsers/FinancialDataHttpParser.scala [hunk 3]
```diff
@@ -21,16 +21,17 @@ import models.errors.UnexpectedError
...
+object FinancialDataHttpParser extends LoggingUtil {
 
   implicit object DirectDebitReads extends HttpReads[HttpResult[DirectDebit]] {
     override def read(method: String, url: String, response: HttpResponse): HttpResult[DirectDebit] = {
+      implicit val res: HttpResponse = response
       response.status match {
         case OK => Right(response.json.as[DirectDebit])
         case status =>
-          logger.warn(
+          errorLogRes(
```
Comment (scott-goodwin) [inline app/connectors/httpParsers/FinancialDataHttpParser.scala:34]: should be a warn

File: app/connectors/httpParsers/VerifyPasscodeHttpParser.scala [hunk 4]
```diff
@@ -33,30 +33,32 @@ object VerifyPasscodeHttpParser extends LoggerUtil {
...
         case CREATED =>
-          logger.debug("[VerifyPasscodeHttpParser][VerifyPasscodeHttpReads][read] - Email successfully verified")
+          debug("[VerifyPasscodeHttpParser][VerifyPasscodeHttpReads][read] - Email successfully verified")
           Right(SuccessfullyVerified)
         case NO_CONTENT =>
-          logger.debug("[VerifyPasscodeHttpParser][VerifyPasscodeHttpReads][read] - Email is already verified")
+          warnLogRes("[VerifyPasscodeHttpParser][VerifyPasscodeHttpReads][read] - Email is already verified")
           Right(AlreadyVerified)
         case FORBIDDEN =>
-          logger.debug("[VerifyPasscodeHttpParser][VerifyPasscodeHttpReads][read] - Max attempts per session exceeded")
```
Comment (scott-goodwin) [inline app/connectors/httpParsers/VerifyPasscodeHttpParser.scala:45]: shoud be a warn

File: app/connectors/httpParsers/VerifyPasscodeHttpParser.scala [hunk 5]
```diff
@@ -33,30 +33,32 @@ object VerifyPasscodeHttpParser extends LoggerUtil {
...
+          errorLogRes("[VerifyPasscodeHttpParser][VerifyPasscodeHttpReads][read] - Max attempts per session exceeded")
           Right(TooManyAttempts)
         case NOT_FOUND if response.body.contains("PASSCODE_NOT_FOUND") =>
-          logger.warn("[VerifyPasscodeHttpParser][VerifyPasscodeHttpReads][read] - Passcode not found (or expired) for this email")
```
Comment (scott-goodwin) [inline app/connectors/httpParsers/VerifyPasscodeHttpParser.scala:48]: should be a warn

File: app/controllers/BaseController.scala [hunk 6]
```diff
@@ -18,33 +18,33 @@ package controllers
...
             None
         }
       } else {
-        logger.info("[BaseController][extractRedirectUrl] couldn't create ContinueUrl from empty string.")
+        warnLog("[BaseController][extractRedirectUrl] couldn't create ContinueUrl from empty string.")
         None
       }
     } catch {
       case e: Exception =>
-        logger.warn("[BaseController][extractRedirectUrl] couldn't create ContinueUrl from what was provided.", e)
```
Comment (scott-goodwin) [inline app/controllers/BaseController.scala:47]: keep as warn

File: app/controllers/agent/ConfirmClientVrnController.scala [hunk 7]
```diff
@@ -69,10 +70,14 @@ class ConfirmClientVrnController @Inject()(authenticate: AuthoriseAsAgentWithCli
               SessionKeys.clientName -> customerDetails.clientName, SessionKeys.insolventWithoutAccessKey -> "false",
               SessionKeys.mtdVatAgentDDMandateFound -> hasDDSetup)
 
-          case Left(Migration) => PreconditionFailed(accountMigrationView())
-          case Left(NotSignedUp) => NotFound(notSignedUpView())
+          case Left(Migration) =>
+            errorLog("[ConfirmClientVrnController][show] - could not retrieve customer details, account creation not completed")
+            PreconditionFailed(accountMigrationView())
+          case Left(NotSignedUp) =>
+            errorLog("[ConfirmClientVrnController][show] - user not signed up, could not retrieve customer details")
```
Comment (scott-goodwin) [inline app/controllers/agent/ConfirmClientVrnController.scala:77]: warn

File: app/controllers/predicates/AuthoriseAsAgentOnly.scala [hunk 8]
```diff
@@ -52,35 +53,35 @@ class AuthoriseAsAgentOnly @Inject()(enrolmentsAuthService: EnrolmentsAuthServic
...
+            infoLog("[AuthoriseAsAgentOnly][invokeBlock] - Is an Agent, checking HMRC-AS-AGENT enrolment")
             checkAgentEnrolment(allEnrolments, block)
           case (_, _) =>
-            logger.debug("[AuthoriseAsAgentOnly][invokeBlock] - Is NOT an Agent, rendering Technical Difficulties view")
+            errorLog("[AuthoriseAsAgentOnly][invokeBlock] - Is NOT an Agent, rendering Technical Difficulties view")
             Future.successful(errorHandler.showInternalServerError)
         }
       case _ =>
-        logger.warn("[AuthoriseAsAgentOnly][invokeBlock] - Missing affinity group")
+        errorLog("[AuthoriseAsAgentOnly][invokeBlock] - Missing affinity group")
```
Comment (scott-goodwin) [inline app/controllers/predicates/AuthoriseAsAgentOnly.scala:63]: warn

File: app/controllers/predicates/AuthoriseAsAgentOnly.scala [hunk 9]
```diff
@@ -52,35 +53,35 @@ class AuthoriseAsAgentOnly @Inject()(enrolmentsAuthService: EnrolmentsAuthServic
...
         Future.successful(errorHandler.showInternalServerError)
     } recover {
       case _: NoActiveSession =>
-        logger.debug("[AuthoriseAsAgentOnly][invokeBlock] - No Active Session, rendering Session Timeout view")
```
Comment (scott-goodwin) [inline app/controllers/predicates/AuthoriseAsAgentOnly.scala:66]: warn

File: app/controllers/predicates/AuthoriseAsAgentOnly.scala [hunk 10]
```diff
@@ -52,35 +53,35 @@ class AuthoriseAsAgentOnly @Inject()(enrolmentsAuthService: EnrolmentsAuthServic
...
+        errorLog("[AuthoriseAsAgentOnly][invokeBlock] - No Active Session, rendering Session Timeout view")
         Unauthorized(sessionTimeoutView())
       case _: AuthorisationException =>
-        logger.warn("[AuthoriseAsAgentOnly][invokeBlock] - Authorisation Exception, rendering Technical Difficulties view")
+        errorLog("[AuthoriseAsAgentOnly][invokeBlock] - Authorisation Exception, rendering Technical Difficulties view")
         errorHandler.showInternalServerError
       case error: UpstreamErrorResponse =>
-        logger.warn(s"[AuthoriseAsAgentOnly][invokeBlock] - Upstream error response received: ${error.message}")
```
Comment (scott-goodwin) [inline app/controllers/predicates/AuthoriseAsAgentOnly.scala:72]: warn

File: app/controllers/predicates/AuthoriseAsAgentWithClient.scala [hunk 11]
```diff
@@ -53,27 +54,28 @@ class AuthoriseAsAgentWithClient @Inject()(enrolmentsAuthService: EnrolmentsAuth
...
           case None ~ _ =>
+            errorLog("[][] - no enrolment found")
             Future.successful(serviceErrorHandler.showInternalServerError)
           case _ ~ allEnrolments =>
             val agent = Agent(allEnrolments)
             val user = User(vrn, active = true, Some(agent.arn))
             block(user)
         } recover {
           case _: NoActiveSession =>
-            logger.debug("[AuthoriseAsAgentWithClient][invokeBlock] - Agent does not have an active session, rendering Session Timeout")
```
Comment (scott-goodwin) [inline app/controllers/predicates/AuthoriseAsAgentWithClient.scala:66]: warn

File: app/controllers/predicates/AuthoriseAsAgentWithClient.scala [hunk 12]
```diff
@@ -53,27 +54,28 @@ class AuthoriseAsAgentWithClient @Inject()(enrolmentsAuthService: EnrolmentsAuth
...
+            errorLog("[AuthoriseAsAgentWithClient][invokeBlock] - Agent does not have an active session, rendering Session Timeout")
             Unauthorized(sessionTimeoutView())
           case _: AuthorisationException =>
-            logger.debug("[AuthoriseAsAgentWithClient][invokeBlock] - Agent does not have delegated authority for Client")
```
Comment (scott-goodwin) [inline app/controllers/predicates/AuthoriseAsAgentWithClient.scala:69]: warn

File: app/connectors/httpParsers/CustomerDetailsHttpParser.scala (hunk 1 above)
Comment (derickdamoah) [reply app/connectors/httpParsers/CustomerDetailsHttpParser.scala:37]: this does not contain any PII so it should bee fine

File: app/connectors/httpParsers/CustomerDetailsHttpParser.scala [hunk 13]
```diff
@@ -21,35 +21,36 @@ import models.CustomerDetails
...
             invalid => {
-              logger.debug(s"[CustomerCircumstancesHttpParser][read]: Invalid Json - $invalid")
+              warnLogRes(s"[CustomerCircumstancesHttpParser][read]: Invalid Json - $invalid")
               Left(UnexpectedError(INTERNAL_SERVER_ERROR, "Invalid Json"))
             },
             valid => Right(valid)
           )
 
         case PRECONDITION_FAILED =>
-          logger.debug(s"[CustomerCircumstancesHttpParser][read]: Status $PRECONDITION_FAILED")
```
[the remark this answers is not in the document]
Comment (derickdamoah) [reply app/connectors/httpParsers/CustomerDetailsHttpParser.scala:44]: changed to warn

File: app/connectors/httpParsers/CustomerDetailsHttpParser.scala (hunk 2 above)
Comment (derickdamoah) [reply app/connectors/httpParsers/CustomerDetailsHttpParser.scala:48]: changed to warn

File: app/connectors/httpParsers/VerifyPasscodeHttpParser.scala (hunk 4 above)
Comment (derickdamoah) [reply app/connectors/httpParsers/VerifyPasscodeHttpParser.scala:45]: changed to warnLog

File: app/connectors/httpParsers/VerifyPasscodeHttpParser.scala (hunk 5 above)
Comment (derickdamoah) [reply app/connectors/httpParsers/VerifyPasscodeHttpParser.scala:48]: changed to warnLog

File: app/connectors/httpParsers/VerifyPasscodeHttpParser.scala [hunk 14]
```diff
@@ -33,30 +33,32 @@ object VerifyPasscodeHttpParser extends LoggerUtil {
...
+          errorLogRes("[VerifyPasscodeHttpParser][VerifyPasscodeHttpReads][read] - Passcode not found (or expired) for this email")
           Right(PasscodeNotFound)
         case NOT_FOUND if response.body.contains("PASSCODE_MISMATCH") =>
-          logger.debug("[VerifyPasscodeHttpParser][VerifyPasscodeHttpReads][read] - Incorrect passcode")
```
[the remark this answers is not in the document]
Comment (derickdamoah) [reply app/connectors/httpParsers/VerifyPasscodeHttpParser.scala:51]: changed to warnLog

File: app/controllers/BaseController.scala (hunk 6 above)
Comment (derickdamoah) [reply app/controllers/BaseController.scala:47]: why should warnLog be used for exceptions?

File: app/controllers/agent/ConfirmClientVrnController.scala [hunk 15]
```diff
@@ -69,10 +70,14 @@ class ConfirmClientVrnController @Inject()(authenticate: AuthoriseAsAgentWithCli
               SessionKeys.clientName -> customerDetails.clientName, SessionKeys.insolventWithoutAccessKey -> "false",
               SessionKeys.mtdVatAgentDDMandateFound -> hasDDSetup)
 
-          case Left(Migration) => PreconditionFailed(accountMigrationView())
-          case Left(NotSignedUp) => NotFound(notSignedUpView())
+          case Left(Migration) =>
+            errorLog("[ConfirmClientVrnController][show] - could not retrieve customer details, account creation not completed")
```
[the remark this answers is not in the document]
Comment (derickdamoah) [reply app/controllers/agent/ConfirmClientVrnController.scala:74]: changed to warnLog

File: app/controllers/predicates/AuthoriseAsAgentOnly.scala (hunk 8 above)
Comment (derickdamoah) [reply app/controllers/predicates/AuthoriseAsAgentOnly.scala:63]: why should this be warn? its throwing an ISE

File: app/controllers/predicates/AuthoriseAsAgentOnly.scala (hunk 9 above)
Comment (derickdamoah) [reply app/controllers/predicates/AuthoriseAsAgentOnly.scala:66]: changed to warnLog

File: app/controllers/predicates/AuthoriseAsAgentOnly.scala [hunk 16]
```diff
@@ -52,35 +53,35 @@ class AuthoriseAsAgentOnly @Inject()(enrolmentsAuthService: EnrolmentsAuthServic
...
-        logger.warn("[AuthoriseAsAgentOnly][invokeBlock] - Missing affinity group")
+        errorLog("[AuthoriseAsAgentOnly][invokeBlock] - Missing affinity group")
         Future.successful(errorHandler.showInternalServerError)
     } recover {
       case _: NoActiveSession =>
-        logger.debug("[AuthoriseAsAgentOnly][invokeBlock] - No Active Session, rendering Session Timeout view")
+        errorLog("[AuthoriseAsAgentOnly][invokeBlock] - No Active Session, rendering Session Timeout view")
         Unauthorized(sessionTimeoutView())
       case _: AuthorisationException =>
-        logger.warn("[AuthoriseAsAgentOnly][invokeBlock] - Authorisation Exception, rendering Technical Difficulties view")
```
[the remark this answers is not in the document]
Comment (derickdamoah) [reply app/controllers/predicates/AuthoriseAsAgentOnly.scala:69]: why should this be warn? it throws an authorisation exception

File: app/controllers/predicates/AuthoriseAsAgentOnly.scala [hunk 17]
```diff
@@ -52,35 +53,35 @@ class AuthoriseAsAgentOnly @Inject()(enrolmentsAuthService: EnrolmentsAuthServic
...
   }
 
   private def checkAgentEnrolment[A](enrolments: Enrolments, block: Agent[A] => Future[Result])(implicit request: Request[A]) =
     if (enrolments.enrolments.exists(_.key == EnrolmentKeys.agentEnrolmentId)) {
-      logger.debug("[AuthoriseAsAgentOnly][checkAgentEnrolment] - Authenticated as agent")
+      debug("[AuthoriseAsAgentOnly][checkAgentEnrolment] - Authenticated as agent")
       block(Agent(enrolments))
     }
     else {
-      logger.debug(s"[AuthoriseAsAgentOnly][checkAgentEnrolment] - Agent without HMRC-AS-AGENT enrolment. Enrolments: $enrolments")
```
[the remark this answers is not in the document]
Comment (derickdamoah) [reply app/controllers/predicates/AuthoriseAsAgentOnly.scala:83]: why should this be warn? the agent doesnt have any enrolment

File: app/controllers/predicates/AuthoriseAsAgentOnly.scala (hunk 10 above)
Comment (derickdamoah) [reply app/controllers/predicates/AuthoriseAsAgentOnly.scala:72]: why should this be warn? its an upstream error

File: app/controllers/predicates/AuthoriseAsAgentWithClient.scala (hunk 11 above)
Comment (derickdamoah) [reply app/controllers/predicates/AuthoriseAsAgentWithClient.scala:66]: changed to warnLog

File: app/controllers/predicates/AuthoriseAsAgentWithClient.scala (hunk 12 above)
Comment (derickdamoah) [reply app/controllers/predicates/AuthoriseAsAgentWithClient.scala:69]: why is this warn? its an authorisation error

**Pull request merged**